framework/sandstone.cpp

```diff
@@ -418,24 +418,34 @@ static void __attribute__((noreturn)) report_fail_common()
 
 void _report_fail(const struct test *test, const char *file, int line)
 {
+#ifndef SANDSTONE_IGNORE_TEST_FAILS
```

Comment:
Is there a technical reason why we're using compiler seams (`#ifdefs`)?

If not, it'd be best not to have to recompile to get the support as this would cause addresses to shift which could change test and framework behavior.

Reply:
See @thiagomacieira comment: https://github.com/opendcdiag/opendcdiag/pull/473#discussion_r1479186986 

If we want to have a runtime option we'll need patterns like this:

```
#define report_fail(test) \
    if (check_if_test_failure_ignored()) { \
        _report_fail_ignored(test, __FILE__, __LINE__); \
    } else { \
        _report_fail(test, __FILE__, __LINE__); \
    }
```